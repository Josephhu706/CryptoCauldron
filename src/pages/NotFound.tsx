import React from "react";
import { ThemeProvider, createTheme } from "@mui/material/styles";
import { CssBaseline } from "@mui/material/";
import notFound from "../assets/images/404.svg";
import "../styles/notFoundPage.scss";
import Button from "@mui/material/Button";
import ArrowBackIosIcon from "@mui/icons-material/ArrowBackIos";
import { useNavigate } from "react-router-dom";

const darkTheme = createTheme({
  palette: {
    mode: "dark",
  },
});

const NotFound = () => {
  const navigate = useNavigate();
  return (
    <div data-testid="notFound" className="notFound container">
      <ThemeProvider theme={darkTheme}>
        <CssBaseline />
        <img src={notFound} alt="" />
        <Button
          data-testid="backButton"
          className="backButton"
          variant="outlined"
          onClick={() => navigate("/")}
        >
          <ArrowBackIosIcon /> Back To Home
        </Button>
      </ThemeProvider>
    </div>
  );
};

export default NotFound;
