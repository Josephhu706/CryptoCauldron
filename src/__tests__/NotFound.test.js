import {
  render,
  screen,
  cleanup,
  waitForElement,
  getByTestId,
  fireEvent,
  act,
} from "@testing-library/react";
import "@testing-library/jest-dom";
import NotFound from "../pages/NotFound";
import renderer from "react-test-renderer";

afterEach(() => {
  cleanup();
});

const mockedUsedNavigate = jest.fn();
jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useNavigate: () => mockedUsedNavigate,
}));

test("Not Found component should be rendered", () => {
  act(() => render(<NotFound />));
  const notFound = screen.getByTestId("notFound");
  expect(notFound).toBeInTheDocument();
});

test("clicking Home should navigate to homepage", () => {
  render(<NotFound />);
  const buttonEl = screen.getByTestId("backButton");
  fireEvent.click(buttonEl);
  expect(mockedUsedNavigate).toHaveBeenCalledWith(`/`);
});

test("404 Page matches snapshot", () => {
  const tree = renderer.create(<NotFound />).toJSON();
  expect(tree).toMatchSnapshot();
});
