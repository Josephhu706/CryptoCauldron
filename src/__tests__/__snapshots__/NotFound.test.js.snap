// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`404 Page matches snapshot 1`] = `
<div
  className="notFound container"
  data-testid="notFound"
>
  <img
    alt=""
    src="404.svg"
  />
  <button
    className="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root backButton css-1vbvugf-MuiButtonBase-root-MuiButton-root"
    data-testid="backButton"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="ArrowBackIosIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M11.67 3.87 9.9 2.1 0 12l9.9 9.9 1.77-1.77L3.54 12z"
      />
    </svg>
     Back To Home
  </button>
</div>
`;
